### PR TITLE
Homework 3

### PR DESCRIPTION
**Pull request by kdr1mkf**
Homework 3 seems to be done.

### CODE REVIEW

File: src/test/java/com/epam/tc/hw3/component/CheckBoxBlock.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    public void clickCheckbox(String checkboxTitle) {
+        String checkboxTitleToLowerCase = checkboxTitle.toLowerCase();
+        switch (checkboxTitleToLowerCase) {
+            case "water":
+                clickWaterCheckbox();
+                break;
+            case "wind":
+                clickWindCheckbox();
+                break;
+            default:
```
Comment (ZeroShadows) [inline src/test/java/com/epam/tc/hw3/component/CheckBoxBlock.java:43]: make checkboxTitle enum, in enum case you won't need default section, cuz right now default section looks strange

File: src/test/java/com/epam/tc/hw3/component/DropdownColorMenu.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        return new Select(dropdownMenu).getFirstSelectedOption().getText();
+    }
+
+    public void clickOption(String optionText) {
+        String optionTextToLowerCase = optionText.toLowerCase();
+        switch (optionTextToLowerCase) {
+            case "yellow":
+                clickYellowOption();
+                break;
+            default:
```
Comment (ZeroShadows) [inline src/test/java/com/epam/tc/hw3/component/DropdownColorMenu.java:30]: simular comment for this switch case as for previous.
make optionTest as enum

File: src/test/java/com/epam/tc/hw3/component/LogPanel.java [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+            case "wind":
+                result = getWindMsgQty();
+                break;
+            case "selen":
+                result = getSelenMsgQty();
+                break;
+            case "yellow":
+                result = getYellowMsgQty();
+                break;
+            default:
```
Comment (ZeroShadows) [inline src/test/java/com/epam/tc/hw3/component/LogPanel.java:45]: make key as enum, then you will able to avoid empthy default

**Comment by ZeroShadows** [comment]
general comment:
- remove hw4 files from this pr.
- where are hw3 in test folder?

File: src/test/java/com/epam/tc/hw3/component/CheckBoxBlock.java (hunk 1 above)
Comment (kdr1mkf) [reply src/test/java/com/epam/tc/hw3/component/CheckBoxBlock.java:47]: Added: enum, IllegalArgumentException as default option.

File: src/test/java/com/epam/tc/hw3/component/LogPanel.java (hunk 3 above)
Comment (kdr1mkf) [reply src/test/java/com/epam/tc/hw3/component/LogPanel.java:44]: Added: enum, IllegalArgumentException as default option.

File: src/test/java/com/epam/tc/hw3/component/DropdownColorMenu.java (hunk 2 above)
Comment (kdr1mkf) [reply src/test/java/com/epam/tc/hw3/component/DropdownColorMenu.java:32]: Added: enum, IllegalArgumentException as default option.

File: src/test/java/com/epam/tc/hw3/component/CheckBoxBlock.java [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+        super(driver);
+    }
+
+    public Boolean checkboxIsSelected(String checkboxTitle) {
+        Boolean result;
+        String checkboxTitleToLowerCase = checkboxTitle.toLowerCase();
+        CheckboxKey checkboxKey = CheckboxKey.fromKey(checkboxTitleToLowerCase);
+        switch (checkboxKey) {
+            case WATER:
+                result = waterCheckboxIsSelected();
```
Comment (ZeroShadows) [inline src/test/java/com/epam/tc/hw3/component/CheckBoxBlock.java:26]: you don't need result variable here. 
just return a result of waterCheckboxIsSelected()

File: src/test/java/com/epam/tc/hw3/component/CheckBoxBlock.java [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+                break;
+            case WIND:
+                result = windCheckboxIsSelected();
```
Comment (ZeroShadows) [inline src/test/java/com/epam/tc/hw3/component/CheckBoxBlock.java:29]: same comment here

File: src/test/java/com/epam/tc/hw3/component/LogPanel.java [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+
+    public LogPanel(WebDriver driver) {
+        super(driver);
+    }
+
+    public int getMsgQty(LogMessageKey key) {
+        int result;
+        switch (key) {
+            case WATER:
+                result = getWaterMsgQty();
```
Comment (ZeroShadows) [inline src/test/java/com/epam/tc/hw3/component/LogPanel.java:33]: you don't need a variable result.
just return the getWaterMsgQty() result.
Same for other cases.

File: src/test/java/com/epam/tc/hw3/component/LogPanel.java [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+                throw new IllegalStateException();
+        }
+        return result;
+    }
+
+    public Set<String> getLogMsgs(LogMessageKey key) {
+        Set<String> result;
+        switch (key) {
+            case WATER:
+                result = getWaterMsgTexts();
```
Comment (ZeroShadows) [inline src/test/java/com/epam/tc/hw3/component/LogPanel.java:54]: you don't need a variable result.
just return the getWaterMsgTexts() result.
Same for other cases.

File: src/test/java/com/epam/tc/hw3/util/CheckboxKey.java [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+        this.key = key;
+    }
+
+    static {
+        for (CheckboxKey checkBoxKey : CheckboxKey.values()) {
+            lookup.put(checkBoxKey.getKey(), checkBoxKey);
+        }
+    }
+
+    public static CheckboxKey fromKey(String key) {
```
Comment (ZeroShadows) [inline src/test/java/com/epam/tc/hw3/util/CheckboxKey.java:24]: I'd like to see iteration through the values() and comparing with key.
without any hashmaps and static block.
Send an example to you in teams

Comment (ZeroShadows) [reply src/test/java/com/epam/tc/hw3/util/CheckboxKey.java:24]: also its true for other enums

File: src/test/java/com/epam/tc/hw3/component/CheckBoxBlock.java (hunk 4 above)
Comment (kdr1mkf) [reply src/test/java/com/epam/tc/hw3/component/CheckBoxBlock.java]: Fixed.

File: src/test/java/com/epam/tc/hw3/component/LogPanel.java (hunk 6 above)
Comment (kdr1mkf) [reply src/test/java/com/epam/tc/hw3/component/LogPanel.java]: Fixed.

File: src/test/java/com/epam/tc/hw3/component/CheckBoxBlock.java (hunk 5 above)
Comment (kdr1mkf) [reply src/test/java/com/epam/tc/hw3/component/CheckBoxBlock.java]: Fixed.

File: src/test/java/com/epam/tc/hw3/component/LogPanel.java (hunk 7 above)
Comment (kdr1mkf) [reply src/test/java/com/epam/tc/hw3/component/LogPanel.java]: Fixed.

File: src/test/java/com/epam/tc/hw3/util/CheckboxKey.java (hunk 8 above)
Comment (kdr1mkf) [reply src/test/java/com/epam/tc/hw3/util/CheckboxKey.java:14]: Fixed.

**Review by ZeroShadows — APPROVED** [review]

**Pull request merged**